meta-mbl-distro/scripts/create-update-payload

```diff
@@ -0,0 +1,214 @@
+can't resolve the paths it needs.
+"""
+
+import pathlib
+import re
+import shutil
+import sys
+import tarfile
+import tempfile
+
+
+# Set the PYTHONPATH so we can import openembedded scripts/lib modules.
+oe_scripts_path = (
+    pathlib.Path(__file__).resolve().parent.parent.parent.parent
+    / "openembedded-core"
+    / "scripts"
+)
+
+lib_path = oe_scripts_path / "lib"
+sys.path.insert(0, str(lib_path))
+
+import scriptpath
+import scriptutils
+
+# This is needed so we can import bitbake modules.
+bitbakepath = scriptpath.add_bitbake_lib_path()
+
+import bb.tinfoil
+import argparse_oe
+
+
+def parse_args():
+    """Parse the command line arguments."""
+    parser = argparse_oe.ArgumentParser()
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "-f",
+        "--filenames",
+        nargs="+",
+        help="Space separated list of filenames of update components.",
+    )
+    group.add_argument(
+        "-b",
+        "--bootloader-slots",
+        nargs="+",
+        help="Space separated list of bootloader slot numbers. e.g 1 would"
+        " correspond to bootloader slot one.",
+    )
+    parser.add_argument(
+        "-o",
+        "--tar-output-path",
+        type=lambda x: pathlib.Path(x).absolute(),
+        default="payload.tar",
+        help="Output path for tar archive containing the update payload.",
+    )
+    args = parser.parse_args()
+
+    return args
+
+
+def match_varnames_to(filenames, tinfoil=None):
+    """
+    Match bitbake variable names to the given file names.
+
+    Parse the bitbake config data using the tinfoil API and find the
+    variable which is bound to the file name.
+
+    Return a dict with the variable names as keys and
+    file names as values.
+
+    :param filenames list: List of file names to look for.
+    :param tinfoil Tinfoil: bitbake tinfoil object.
+    """
+    output_map = {}
+    for cvar in tinfoil.config_data:
+        if cvar.startswith("MBL_WKS"):
+            val = tinfoil.config_data.getVar(cvar).strip()
+            if val in filenames:
+                output_map[cvar.split("_FILENAME")[0]] = val
+
+    if not output_map:
+        raise FileNotFoundError(
+            "No matching vars were found."
+            "\nThe available config data is: {}"
+            "\n\nThe filenames we are trying to match are: {}".format(
+                list(tinfoil.config_data), filenames
+            )
+        )
+    return output_map
+
+
+def match_filenames_to(bootloader_slots, tinfoil=None):
+    """
+    Look up the bitbake variables corresponding to the bootloader slots.
+
+    Parse the bitbake config data using the tinfoil API and match the
+    MBL_WKS_BOOTLOADER variable to the given bootloader slot.
+
+    Return a dict with the variable name as the key and the file name as the
+    value.
+
+    :param bootloader_slots list: List of bootloader slots e.g [1, 2]
```

Comment:
Ah, I notice you've removed "fs" from that list. I was just thinking that we can't handle "fs" like the others because it doesn't have a `_FILENAME` variable - it uses OE's BOOTFILES mechanism.